File: ios/LensWarmup.m
#import <React/RCTBridgeModule.h>

// Bridges the Swift LensWarmupModule (preloads the MediaPipe face model at app start).
@interface RCT_EXTERN_MODULE(LensWarmup, NSObject)
RCT_EXTERN_METHOD(warmUp:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/LensWarmup.swift
import Foundation
import React

// Preloads the shared MediaPipe FaceLandmarker off the main thread, so the first lens
// selection doesn't stutter on the model load. Called from JS on app start.
@objc(LensWarmup)
class LensWarmup: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool { false }

  @objc(warmUp:rejecter:)
  func warmUp(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    DispatchQueue.global(qos: .utility).async {
      FaceLandmarkerShared.shared.warmUp()
      resolve(nil)
    }
  }
}
